packages/components/src/navigation/item/use-navigation-tree-item.tsx

```diff
@@ -22,7 +27,8 @@ export const useNavigationTreeItem = ( itemId, props ) => {
 	useEffect( () => {
 		const isMenuActive = activeMenu === menu;
 		const isItemVisible =
-			! search || normalizedSearch( props.title, search );
+			! search ||
+			( !! props.title && normalizedSearch( props.title, search ) );
```

Comment:
⚠️ Runtime change:

Since it doesn't really make sense for `normalizedSearch` to accept `undefined` or empty string values, I've added a check on `props.title`.

Only doubt is: should we expect `normalizedSearch( '', '' )` and/or `normalizedSearch( undefined, '' )` to return `true` ?